File: sdk/python/examples/memory_search.py
"""
Tortoise Python SDK - Memory Management Example
"""

import asyncio
from tortoise import TortoiseClient


async def main():
    async with TortoiseClient(
        gateway_url="http://localhost:8080",
        api_key="your-api-key"
    ) as client:
        # Add memories
        print("Adding memories...\n")
        
        await client.memory_add(
            content="User prefers to be called 'Alex'",
            memory_type="user_preference"
        )
        await client.memory_add(
            content="User is interested in Python programming",
            memory_type="interest"
        )
        await client.memory_add(
            content="User works as a software engineer at a startup",
            memory_type="work"
        )
        
        print("Added 3 memories\n")
        
        # Search memories
        print("Searching for 'programming'...\n")
        results = await client.memory_search(query="programming")
        
        for entry in results:
            print(f"- {entry['content']} ({entry['type']})")


if __name__ == "__main__":
    asyncio.run(main())

File: sdk/rust/examples/memory_search.rs
//! Tortoise Rust SDK - Memory Management Example

use tortoise_sdk::{Client, ClientConfig};

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let config = ClientConfig::default()
        .with_url("http://localhost:8080")
        .with_api_key("your-api-key");
    
    let client = Client::new(config);
    
    println!("Adding memories...\n");
    
    // Add memories
    client.memory_add(
        "User prefers dark mode interface",
        "user_preference"
    ).await?;
    
    client.memory_add(
        "User is interested in AI and machine learning",
        "interest"
    ).await?;
    
    client.memory_add(
        "User works as a backend developer",
        "work"
    ).await?;
    
    println!("Added 3 memories\n");
    
    // Search memories
    println!("Searching for 'AI'...");
    let results = client.memory_search("AI", 10).await?;
    
    for entry in results {
        println!("- {} ({})", entry.content, entry.memory_type);
    }
    
    println!("\nDone!");
    Ok(())
}
